File: Programmers/dnjfqhd12345/week_1/최소직사각형.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    int arrsize = sizes.size();
    for (int i = 0; i < arrsize; i++) {
        sort(sizes[i].begin(), sizes[i].end());
    }
    cout << sizes[0][0] << sizes[0][1] << endl;
    int min = sizes[0][0];
    int max = sizes[0][1];
    for (int i = 0; i < arrsize; i++) {
        if (sizes[i][0] >= min)
            min = sizes[i][0];
    }
    for (int i = 0; i < arrsize; i++) {
        if (sizes[i][1] >= max)
            max = sizes[i][1];
    }
    answer = min * max;
    return answer;
}

File: Programmers/moonmonoon/week_5/최소직사각형.py
def solution(sizes):
    answer = 0
    for s in sizes:
        if s[0] < s[1]:
            temp = s[0]
            s[0] = s[1]
            s[1] = temp

    a = [a[0] for a in sizes]
    b = [b[1] for b in sizes]

    answer = max(a) * max(b)
    return answer
